src/controllers/createpdf.controller.ts

```diff
@@ -0,0 +1,92 @@
+import { Controller, Get, Res, Post, Body, Render, Param, Query} from '@nestjs/common';
+import {Response} from 'express';
+import {createUserData, createLogInfo,createRunConf} from '../dtos/createpdf.dto';
+import {CreatePdfService} from "../services/createpdf.service";
+import { Log } from '../entities/log.entity';
+import { getManager } from 'typeorm';
+import { Run } from '../entities/run.entity';
+import { ApiUseTags, ApiBearerAuth } from '@nestjs/swagger';
+
+@ApiUseTags('createpdf')
+@Controller('createpdf')
+export class CreatepdfController {
+  constructor(public createPdfService: CreatePdfService){};
+
+  @Get('getlog')
+  async getLogInfo(@Query() query: createLogInfo, @Res() res: Response): Promise<void>{
+    var allDate = this.createPdfService.getDate();
+
+    const repository = getManager().getRepository(Log);
+    
+    const result = await repository.createQueryBuilder("log")
+      .leftJoinAndSelect("log.user", "user_id, external_id, sams_id")
+      .where("log.log_id = :id", {id: query.id})
+      .getOne();
+
+    let params = {user:result["user"]["userId"], external:result["user"]["externalUserId"], sams:result["user"]["samsId"], logId:result["logId"],
```

Comment:
Why are you manually mapping the result? If the log object changes for example, the developer would need to update the params object.